pkg/controller/bundlec/bundle_sync_task.go

```diff
@@ -414,6 +414,41 @@ func (st *bundleSyncTask) handleProcessResult(retriable bool, processErr error)
 		bundleUpdated = updateBundleCondition(st.bundle, &readyCond) || bundleUpdated
 		bundleUpdated = updateBundleCondition(st.bundle, &errorCond) || bundleUpdated
 
+		// Plugin statuses
+		name2status := make(map[smith_v1.PluginName]struct{})
+		// most likely will be of the same size as before
+		pluginStatuses := make([]smith_v1.PluginStatus, 0, len(st.bundle.Status.PluginStatuses))
+		for _, res := range st.bundle.Spec.Resources { // Deterministic iteration order
+			if res.Spec.Plugin == nil {
+				continue // Not a plugin
+			}
+			pluginName := res.Spec.Plugin.Name
+			if _, ok := name2status[pluginName]; ok {
+				continue // Already reported
+			}
+			name2status[pluginName] = struct{}{}
+			var pluginStatus smith_v1.PluginStatus
+			pluginContainer, ok := st.pluginContainers[pluginName]
+			if ok {
+				describe := pluginContainer.Plugin.Describe()
```

Comment:
What if plugin returns `nil`?
Probably fine to explode in that case, rather than continue running Smith with invalid plugin.

Reply:
Hm, I wonder why it is a pointer in the first place? Should probably change it to a non-pointer.
We call this function in a lot of places and never assume it can return a `nil` pointer.

Reply:
I'll send a follow up to remove the pointer here.

Reply:
Or maybe not. Not sure about this one....